### PR TITLE
GroupState: add serialization support

### PR DESCRIPTION
**Pull request by akva2**
Sits on top of https://github.com/OPM/opm-simulators/pull/4437
Downstream of https://github.com/OPM/opm-common/pull/3392

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this opm-models=780 opm-common=3392 please

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

**Review by atgeirr — COMMENTED** [review]
The changes look good. But I noticed that the equality operator is not comparing gpmaint_state, probably overlooked when that was added earlier!

**Comment by akva2** [comment]
nice catch. addressed here and in https://github.com/OPM/opm-common/pull/3402

**Comment by akva2** [comment]
jenkins build this opm-common=3402 please

File: opm/simulators/wells/WellContainer.hpp [hunk 1]
```diff
@@ -45,15 +45,27 @@ namespace Opm {
 template <class T>
 class WellContainer {
 public:
-
     WellContainer() = default;
 
+    WellContainer(std::size_t size)
+        : m_data(size)
+    {}
```
Comment (bska) [inline opm/simulators/wells/WellContainer.hpp:52]: Would you mind marking this as `explicit` please?

**Review by bska — COMMENTED** [review]
This looks good for the most part, but I think I'd prefer an `explicit` constructor if at all possible.

File: opm/simulators/wells/WellContainer.hpp (hunk 1 above)
Comment (akva2) [reply opm/simulators/wells/WellContainer.hpp]: i'll do even better. i forgot to update my git on the laptop when i did the gpmaint_state fix and thus pushed an old version of this branch. this constructor is in fact not needed at all, and the groupstate one can be defaulted.

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
We might also consider adding a test for something like
```C++
struct X { std::vector<double> x1; std::array<float, 3> x2; };
WellContainer<X>
```
That's a little more representative of the kinds of types we typically stick in the `WellContainer`.  In the meantime I'll merge this into the master branch.

**Pull request merged**